docs/specification.md

```diff
@@ -0,0 +1,169 @@
+## General Specification for the project !   
+> This Specification is only for the MVP of this project that need to do the first user journey , consisting of opening a file editing and saving it into the filesystem !    
+--- 
+> This Specification can't be finished now , since m too retarded to give it soo much time !   
+
+### General Directory of the project !   
```

Comment:
I assume that we should consider using the workspace approach instead of a single binary approach to facilitate the reformation process after the MVP is completed.

Reply:
yes correct , but we will start with one crate , i still lack a global visibility of the usage of various technologies so the mvp will allow us to have a clearer vision and hence the workspace paradigm might be adopted ( or other approaches as well ! )

Reply:
I got it